### DISCUSSION TITLE
Is sugar only for clients

### DISCUSSION THREAD

**Discussion by micttyl** [carbon-language/carbon-lang#2101] [Language design]
Would no syntactic sugar be realized at the parsing layer of Carbon?
In other words, do we only leave this to the interpretation of clients?
In other words, do developers grammatically disregard them in terms of any implementation?
For example, I do not see the arrow expression in `ref_deref_expression` in the yacc parser, which is considered as a 'syntactic sugar' in the [design doc](https://github.com/carbon-language/carbon-lang/tree/trunk/docs/design#pointer-types).

**Comment by chandlerc** [comment]
I think you mean "syntactic sugar"?

> For example, I do not see the arrow expression in `ref_deref_expression` in the yacc parser, which is considered as a 'synthetic sugar' in the [design doc](https://github.com/carbon-language/carbon-lang/tree/trunk/docs/design#pointer-types).

This just hasn't been implemented in the explorer yet.

**Reply by micttyl** [reply]
It's clear. Thanks, I fixed the typo.
